### PR TITLE
memutils introduction proposal.

### PR DESCRIPTION
**Pull request by devnexen**
starting with a scoped callback e.g. when it would become too verbose
to close resources for each error case.

### CODE REVIEW

**Comment by ColinPitrat** [comment]
There's scope_exit in experimental:
https://en.cppreference.com/w/cpp/experimental/scope_exit

I guess we could start using it, not sure how far it is from making it to the standard.

**Comment by devnexen** [comment]
looks interesting but I do not have it in my compilers I have the experimental namespace tough.

**Comment by ColinPitrat** [comment]
I'd be happy to merge this if we rename `deref` to `scope_exit` which seems to be what the standard is currently considering as well as what boost uses.

File: src/cap32.cpp [hunk 1]
```diff
@@ -1220,6 +1221,9 @@ void bin_load (const std::string& filename, const size_t offset)
     return;
   }
 
+  auto closure = [&]() { fclose(file); };
+  memutils::deref<decltype(closure)> cs(closure); // TODO: when C++20, can become a one liner expression.
```
Comment (ColinPitrat) [inline src/cap32.cpp:1225]: What needs to change in C++20 to have the one liner?

Comment (devnexen) [reply src/cap32.cpp]: well with C++ 20 you can simplify like
```
memutils::scope_exit cs([&](){ fclose(file); });
```

**Comment by ColinPitrat** [comment]
Replaced by #236

**Pull request merged**